### PR TITLE
Fix Asset.is_binary_executable

### LINKED ISSUE #41: Asset.is_binary_executable always returns false for relative paths
https://github.com/mmstick/cargo-deb/blob/master/src/manifest.rs#L41

`release_dir_prefix` is an absolute path derived from `cargo_metadata()'s` `manifest.target_path`, which is an absolute path. Thus, `self.source_file.starts_with(release_dir_prefix)` will always return `false` for relative paths specified in `package.metadata.deb.assets`. 

```
rustc --version
rustc 1.23.0 (766bd11c8 2018-01-01)

cargo --version
cargo 0.24.0 (45043115c 2017-12-05)

uname -a
Darwin MacBook-Pro.local 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
```

### PR DESCRIPTION
**Pull request by messense**
Closes #41

### CODE REVIEW

**Comment by kornelski** [comment]
Thanks for the PR. Does it work with cross-compilation? When cross-compiling Cargo doesn't use `target/release`, but `target/<triple>/release`.

**Comment by messense** [comment]
I think not. `workspace_root` will be incorrectly pointed to `target/` directory.

File: src/manifest.rs [hunk 1]
```diff
@@ -37,13 +37,15 @@ impl Asset {
             chmod,
         }
     }
-    pub fn is_binary_executable(&self, release_dir_prefix: &Path) -> bool {
-        self.source_file.starts_with(release_dir_prefix) && 0 != (self.chmod & 0o111)
+    pub fn is_binary_executable(&self, workspace_root: &Path, release_dir_prefix: &Path) -> bool {
+        let source_file_abspath = workspace_root.join(&self.source_file);
```
Comment (messense) [inline src/manifest.rs]: This still needs to rewrite `target/release/xxx` to `target/<triple>release/xxx` I think?

**Comment by messense** [comment]
`workspace_root` seems missing on Travis CI.

Running local with `cargo 0.26.0-nightly (6a8eb71f6 2018-01-13)`:

```bash
$ cargo metadata --format-version=1 | jq .
{
  "packages": [
    {
      "name": "example",
      "version": "0.1.0",
      "id": "example 0.1.0 (path+file:///Users/messense/Projects/cargo-deb/example)",
      "license": null,
      "license_file": null,
      "description": null,
      "source": null,
      "dependencies": [],
      "targets": [
        {
          "kind": [
            "bin"
          ],
          "crate_types": [
            "bin"
          ],
          "name": "example",
          "src_path": "/Users/messense/Projects/cargo-deb/example/src/main.rs"
        }
      ],
      "features": {},
      "manifest_path": "/Users/messense/Projects/cargo-deb/example/Cargo.toml"
    }
  ],
  "workspace_members": [
    "example 0.1.0 (path+file:///Users/messense/Projects/cargo-deb/example)"
  ],
  "resolve": {
    "nodes": [
      {
        "id": "example 0.1.0 (path+file:///Users/messense/Projects/cargo-deb/example)",
        "dependencies": []
      }
    ],
    "root": "example 0.1.0 (path+file:///Users/messense/Projects/cargo-deb/example)"
  },
  "target_directory": "/Users/messense/Projects/cargo-deb/example/target",
  "version": 1,
  "workspace_root": "/Users/messense/Projects/cargo-deb/example"
}
```

**Comment by kornelski** [comment]
Maybe you can use `manifest_path.parent()` instead? or as a fallback?

**Comment by kornelski** [comment]
Superb. Thank you.

**Pull request merged**